pyccel/ast/core.py

```diff
@@ -877,6 +877,48 @@ def __eq__(self, other):
     def __hash__(self):
         return hash((id(self.variable), self.shape, self.order, self.status))
 
+#------------------------------------------------------------------------------
+class Deallocate(Basic):
+    """
+    Represents memory deallocation (usually of an array) for code generation.
+    This is relevant to low-level target languages, such as C or Fortran,
+    where the programmer must take care of heap memory deallocation.
+
+    Parameters
+    ----------
+    variable : pyccel.ast.core.Variable
+        The typed variable (usually an array) that needs memory deallocation.
+
+    Notes
+    -----
+    An object of this class is immutable, although it contains a reference to a
+    mutable Variable object.
```

Comment:
you shouldn't capitalize `variable`

Reply:
I think it's correct to capitalize when we are talking about a class object